packages/dataviews/src/components/dataviews-filters/input-widget.tsx

```diff
@@ -0,0 +1,73 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import type { View, NormalizedFilter, NormalizedField } from '../../types';
+
+interface UserInputWidgetProps {
+	view: View;
+	filter: NormalizedFilter;
+	onChangeView: ( view: View ) => void;
+	fields: NormalizedField< any >[];
+}
+
+export default function InputWidget( {
+	filter,
+	view,
+	onChangeView,
+	fields,
+}: UserInputWidgetProps ) {
+	const currentFilter = view.filters?.find(
+		( f ) => f.field === filter.field
+	);
+	if ( ! currentFilter ) {
+		return null;
+	}
+
+	const field = fields.find( ( f ) => f.id === filter.field );
+	if ( ! field || ! field.Edit ) {
+		return null;
+	}
+
+	const currentValue = currentFilter.value;
+
+	const handleChange = useCallback(
+		( data: Record< string, any > ) => {
+			const nextValue = data[ field.id ];
+			if ( nextValue === currentValue ) {
+				return;
+			}
+
+			onChangeView( {
+				...view,
+				filters: ( view.filters ?? [] ).map( ( _filter ) =>
+					_filter.field === filter.field
+						? {
+								..._filter,
+								operator:
+									currentFilter.operator ||
+									filter.operators[ 0 ],
+								value: nextValue,
+						  }
+						: _filter
+				),
+			} );
+		},
+		[ currentValue, field, onChangeView, view, filter, currentFilter ]
+	);
+
+	return (
+		<div className="dataviews-filters__user-input-widget">
+			<field.Edit
+				data={ { [ field.id ]: currentValue } }
```

Comment:
This Edit function doesn't have access to the whole object, just one value. This is not a restriction we have in DataForm, where the Edit is used. What if the Edit access values other than the `field.id` (e.g., "price" + "discount" to generate a "consolidated price")?

Reply:
I've changed how InputWidget passes data to Edit, providing all filter values instead in 5920eedd86b6b8e52ee98c4939332333fa45b165.